crates/bevy_pbr/src/prepass/mod.rs

```diff
@@ -467,32 +628,93 @@ pub fn extract_camera_prepass_phase(
                 &Camera,
                 Option<&DepthPrepass>,
                 Option<&NormalPrepass>,
+                Option<&MotionVectorPrepass>,
+                Option<&PreviousViewProjection>,
             ),
             With<Camera3d>,
         >,
     >,
 ) {
-    for (entity, camera, depth_prepass, normal_prepass) in cameras_3d.iter() {
-        if !camera.is_active {
-            continue;
-        }
+    for (
+        entity,
+        camera,
+        depth_prepass,
+        normal_prepass,
+        motion_vector_prepass,
+        maybe_previous_view_proj,
+    ) in cameras_3d.iter()
+    {
+        if camera.is_active {
+            let mut entity = commands.get_or_spawn(entity);
+
+            if depth_prepass.is_some()
+                || normal_prepass.is_some()
+                || motion_vector_prepass.is_some()
+            {
+                entity.insert((
+                    RenderPhase::<Opaque3dPrepass>::default(),
+                    RenderPhase::<AlphaMask3dPrepass>::default(),
+                ));
+            }
 
-        let mut entity = commands.get_or_spawn(entity);
-        if depth_prepass.is_some() || normal_prepass.is_some() {
-            entity.insert((
-                RenderPhase::<Opaque3dPrepass>::default(),
-                RenderPhase::<AlphaMask3dPrepass>::default(),
-            ));
-        }
-        if depth_prepass.is_some() {
-            entity.insert(DepthPrepass);
-        }
-        if normal_prepass.is_some() {
-            entity.insert(NormalPrepass);
+            if depth_prepass.is_some() {
+                entity.insert(DepthPrepass);
+            }
+            if normal_prepass.is_some() {
+                entity.insert(NormalPrepass);
+            }
+            if motion_vector_prepass.is_some() {
+                entity.insert(MotionVectorPrepass);
+            }
+
+            if let Some(previous_view) = maybe_previous_view_proj {
+                entity.insert(previous_view.clone());
+            }
         }
     }
 }
 
+#[derive(Resource, Default)]
+pub struct PreviousViewProjectionUniforms {
+    pub uniforms: DynamicUniformBuffer<PreviousViewProjection>,
+}
+
+#[derive(Component)]
+pub struct PreviousViewProjectionUniformOffset {
+    pub offset: u32,
+}
+
+pub fn prepare_previous_view_projection_uniforms(
```

Comment:
Hmm. I wonder if this might be easier just by copying the previous view uniforms buffer rather than involving anything CPU-side...

Reply:
We'd need to keep around the old offsets as well, except components are removed from the render world each frame :/. I could use a resource to copy it in and out, but that's a bit more complicated.